public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -136,31 +135,35 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     };
   }
 
-  getQueryDataSourceIdentifier(): string | null | undefined {
-    const { query, dataSource: dsSettings } = this.props;
-    return query.datasource?.uid ?? dsSettings.uid;
+  getDataSourceIdentifier(props: Props<TQuery>): string | null | undefined {
+    const { query, dataSource } = props;
+
+    return query.datasource?.uid ?? dataSource.uid;
```

Comment:
this is breaking behaviors I think, 

Proposed a simplified logic here: 
https://github.com/grafana/grafana/pull/62617#pullrequestreview-1325773099

Reply:
Thanks for pointing me there!

I see it fixes the interpolated DS issue, I didn't have that into account. However, it doesn't solve the issue of "Where there is no DS with that UI, it should fallback to the default one".
```diff
```

Reply:
I'll bring your proposal to use `rawRef` instead to this PR